Mlw fix release 10.13 10.15 (#114)
SoftwareAG/c8y-releasenotes: content/release-10-13-0/announcements-10-13-0.md

  This is because setting a small `pageSize` without setting `currentPage` can result in queries that run very slowly.
  For example, to request the first 20 items, you must set `pageSize` to 20 and `currentPage` to 1.
  See also [Paging Cumulocity IoT queries](https://documentation.softwareag.com/apama/v10-11/apama10-11/apama-webhelp/index.html#page/apama-webhelp%2Fco-ConApaAppToExtCom_cumulocity_paging_cumulocity_queries.html) in the Apama documentation.

### Machine learning Workbench

#### Implemented

- Currently both "Machine Learning Manager" & "Machine Learning Admin" user groups have access to execute the python scripts, run python commands using jupyter notebook, train workflows & train neural network models.
To further enhance the security, the above functionalities will be limited only to "Machine Learning Admin" user group.

- "Machine Learning Admin" user group should be treated as privileged user group.

> - This access limitation will be a breaking change for existing "Machine Learning Manager" user group.

Reviewer: ```suggestion
- This access limitation will be a breaking change for the existing "Machine Learning Manager" user group.
```

> - This access limitation will be a breaking change for existing "Machine Learning Manager" user group.

Reviewer: I suggest to provide some recommendation to the users what to do

Author: done

> - "Machine Learning Manager" user group has been restricted from accessing below post apis:

Reviewer: ```suggestion
- The "Machine Learning Manager" user group has been restricted from accessing the following post APIs:
```

>     ##### In Training Workflow:

Reviewer: ```suggestion
In training workflows:
```

    ```
    {{url}}/service/mlw/projects/{{projectID}}/resources/{{resourcesID}}/workflow
    ```

>     ##### In Code Execution:

Reviewer: ```suggestion
In code execution:
```

    ```
    {{url}}/service/mlw/projects/{{projectID}}/resources/{{resourcesID}}/execute
    ```

>     ##### In NeuralNetwork:

Reviewer: ```suggestion
In neural networks:
```

    ```
    {{url}}/service/mlw/projects/{{projectID}}/resources/{{resourcesID}}/trainNN
    ```
    

> - Below apis are only accessible to  "Machine Learning Admin" user group:

Reviewer: ```suggestion
- The following APIs are only accessible to  the "Machine Learning Admin" user group:
```

>     ##### GET - jnb-sessions:

Reviewer: ```suggestion
GET - jnb-sessions:
```

    ```
    {{url}}/service/mlw/jnb-sessions
    ```

>     ##### GET - jnb-content:

Reviewer: ```suggestion
GET - jnb-content:
```

    ```
    {{url}}/service/mlw/projects/{{projectID}}/resources/{{resourceID}}/jnb-content
    ```

>     ##### PUT - jnb-content:

Reviewer: ```suggestion
PUT - jnb-content:
```

    ```
    {{url}}/service/mlw/projects/{{projectID}}/resources/{{resourceID}}/jnb-content
    ```

> - "Machine Learning Manager" user group has been restricted from executing jupyter notebooks & accessing assets section.

Reviewer: ```suggestion
- The "Machine Learning Manager" user group has been restricted from executing Jupyter notebooks and accessing the assets section.
```

> - "Machine Learning User" user group has been restricted from accessing assets section.

Reviewer: ```suggestion
- The "Machine Learning User" user group has been restricted from accessing the assets section.
```
